Update to latest Microsoft.Extensions.AI version

eerhardt opened the pull request:
In preparation for the next release, which is currently scheduled to be released right before Aspire 9.2

## Checklist

- Is this feature complete?
  - [x] Yes. Ready to ship.
- Are you including unit tests for the changes and scenario tests if relevant?
  - [x] Yes
- Did you add public API?
  - [x] No
- Does the change make any security assumptions or guarantees?
  - [x] No
- Does the change require an update in our Aspire docs?
  - [x] No

eerhardt committed:
Update to latest Microsoft.Extensions.AI version

In preparation for the next release.

eerhardt committed:
Update Azure OpenAI ConfigurationSchema

joperezr commented on eng/Versions.props:
```diff
@@ -34,7 +34,7 @@
     <MicrosoftDotNetBuildTasksWorkloadsVersion>9.0.0-beta.25164.2</MicrosoftDotNetBuildTasksWorkloadsVersion>
     <!-- dotnet/extensions -->
-    <MicrosoftExtensionsAIVersion>9.3.0-preview.1.25161.3</MicrosoftExtensionsAIVersion>
+    <MicrosoftExtensionsAIVersion>9.4.0-preview.1.25201.1</MicrosoftExtensionsAIVersion>
```
Do you mind if I push a change to your branch to make this work with Dependency flow so these versions actually come from that instead? This way we can make it so all of extensions use the same version, and not just MEAI.

eerhardt replied:
I don't mind at all. Please do!

eerhardt replied:
I assume that will mean also updating the rest of the packages from dotnet/extensions? Or are we able to take different versions?

eerhardt replied:
As discussed offline, let's not do this right now. Once Microsoft.Extensions.AI goes stable and ships regularly with the rest of the dotnet/extensions packages, this can be done.

joperezr reviewed (commented):
Changes LGTM, but let me know if it's ok for me to push that extra change here to onboard to dependency flow.

eerhardt committed:
Update MEAIPackageOverrides.targets

- Use the MSBuild properties for the versions in Versions.props. This avoids having to manually update these in the future
- Use PackageVersion instead of PackageReference

joperezr reviewed (approved):
Thanks @eerhardt!
